### PR TITLE
Intake1

### PR DESCRIPTION
**Pull request by 33Uri33**

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class IntakeSubsystem extends SubsystemBase implements IntakeConstants{
+  private TalonFX _motor;
+  private DigitalInput _limitSwitch;
+  //singelton
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:22]: add `private TorqueCurrentFOC currentFOC = new TorqueCurrentFOC(0);`

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:1]: add an set current using `motor.setControl(currentFOC.withOutput(double))`

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  public Command stopIntakeCommand(){
+    return runOnce(() -> _motor.stopMotor());
+  }
+
+  public Command coolectUntilNoteCommand(){
+    return startEnd(() -> setSpeedCommand(),() -> stopIntakeCommand()).until(() -> getSwitchCommand());
+  }
+
+  public Command feedShooterCommand(){
+    return runOnce(() -> _motor.set(FEED_INTAKE_SPEED)).andThen(Commands.waitSeconds(0.2)
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java]: switch the `andThen` to withTimeout and use constants

**Review by eilonhuppert — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java (hunk 2 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java (hunk 1 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:22]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:1]: V (אני לא 100% בטוח שעשיתי את זה נכון)

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  public Command feedShooterCommand(){
+    return startEnd(() -> _motor.setControl(currentFOC.withOutput(FEED_INTAKE_SPEED)),() -> stopIntakeCommand()).withTimeout(FEED_WAIT_TIME)
+    .andThen(stopIntakeCommand());
+  }
+
+  public Command feedShooterCommand(double speed){
+    return startEnd(() -> _motor.setControl(currentFOC.withOutput(speed)),() -> stopIntakeCommand()).withTimeout(FEED_WAIT_TIME)
+    .andThen(stopIntakeCommand());
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java]: you don't need to stop the motor twice

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    return runOnce(() -> _motor.stopMotor());
+  }
+
+  public Command coolectUntilNoteCommand(){
+    return startEnd(() -> setSpeedCommand(),() -> stopIntakeCommand()).until(() -> getSwitchCommand());
+  }
+
+  public Command feedShooterCommand(){
+    return startEnd(() -> _motor.setControl(currentFOC.withOutput(FEED_INTAKE_SPEED)),() -> stopIntakeCommand()).withTimeout(FEED_WAIT_TIME)
+    .andThen(stopIntakeCommand());
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java]: you don't need to stop the motor twice

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+  /**
+   * sets the current to 20 amp
+   * 
+   */
+  public Command setSpeedCommand(){
+    return runOnce(() -> _motor.setControl(currentFOC.withOutput(COLLECT_CURRENT)));
+  }
+
+  public Command setSpeedCommand(double speed){
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java]: change `speed `to `current`

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+  public boolean getSwitchCommand(){
+    return _limitSwitch.get();
+  }
+
+  /**
+   * sets the current to 20 amp
+   * 
+   */
+  public Command setSpeedCommand(){
+    return runOnce(() -> _motor.setControl(currentFOC.withOutput(COLLECT_CURRENT)));
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:51]: change speed to current

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:1]: add javadocs and comments where you think you need them

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+public interface IntakeConstants {
+    int MOTOR_ID = 0; //TODO: find ID
+    int LIMIT_SWITCH_ID = 0;
+    double PEAK_CURRENT = 40.0;
+    NeutralModeValue NEUTRAL_MODE = NeutralModeValue.Coast;
+    InvertedValue INVERTED = InvertedValue.Clockwise_Positive; //check the correct direction
+    double COLLECT_CURRENT = 20;
+    double FEED_INTAKE_SPEED = 0.7;
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: change to current values and names

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    double FEED_WAIT_TIME = 0.2;
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: כנ"ל

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package frc.robot.subsystems.intake;
+
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+public interface IntakeConstants {
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java:6]: add TODO where you not sure what value is there

**Review by eilonhuppert — CHANGES_REQUESTED** [review]
it may seems like a lot but it nothing

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 10]
```diff
@@ -6,10 +6,10 @@
...
     int LIMIT_SWITCH_ID = 0;
-    double PEAK_CURRENT = 40.0;
+    double PEAK_CURRENT = 40.0; //TODO: find ID
     NeutralModeValue NEUTRAL_MODE = NeutralModeValue.Coast;
     InvertedValue INVERTED = InvertedValue.Clockwise_Positive; //check the correct direction
-    double COLLECT_CURRENT = 20;
-    double FEED_INTAKE_SPEED = 0.7;
-    double FEED_WAIT_TIME = 0.2;
+    double COLLECT_CURRENT = 20; //TODO: find ID
+    double FEED_INTAKE_CURRENT = 0.7; //TODO: find ID
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Wrong value think why

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 11]
```diff
@@ -6,10 +6,10 @@
...
+    double FEED_WAIT_TIME = 0.2; //TODO: find ID
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: CP dosent allways work

**Review by eilonhuppert — CHANGES_REQUESTED** [review]

**Comment by Yair4586** [comment]
i think line 111 is unnecessary bc if its getting to this place it didnt break from the configs
you can just delete the if line

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java (hunk 5 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:1]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 7 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 8 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 11 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 12]
```diff
@@ -6,10 +6,10 @@
...
     int MOTOR_ID = 0; //TODO: find ID
     int LIMIT_SWITCH_ID = 0;
-    double PEAK_CURRENT = 40.0;
+    double PEAK_CURRENT = 40.0; //TODO: find ID
     NeutralModeValue NEUTRAL_MODE = NeutralModeValue.Coast;
     InvertedValue INVERTED = InvertedValue.Clockwise_Positive; //check the correct direction
-    double COLLECT_CURRENT = 20;
-    double FEED_INTAKE_SPEED = 0.7;
-    double FEED_WAIT_TIME = 0.2;
+    double COLLECT_CURRENT = 20; //TODO: find ID
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Its not Id its value

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java [hunk 13]
```diff
@@ -6,10 +6,10 @@
 public interface IntakeConstants {
     int MOTOR_ID = 0; //TODO: find ID
     int LIMIT_SWITCH_ID = 0;
-    double PEAK_CURRENT = 40.0;
+    double PEAK_CURRENT = 40.0; //TODO: find ID
```
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Not Id

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 11 above)
Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Not Id

Comment (eilonhuppert) [inline src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: Not Id

**Review by eilonhuppert — CHANGES_REQUESTED** [review]

**Review by eilonhuppert — COMMENTED** [review]
Not Id

**Comment by eilonhuppert** [comment]
> i think line 111 is unnecessary bc if its getting to this place it didnt break from the configs you can just delete the if line

טעות
הברייק הוא ללולאה

**Comment by eilonhuppert** [comment]
עליך לתקן את פליטת הקולמוס שעשית

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 13 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 11 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: V

File: src/main/java/frc/robot/subsystems/intake/IntakeConstants.java (hunk 12 above)
Comment (33Uri33) [reply src/main/java/frc/robot/subsystems/intake/IntakeConstants.java]: V

**Review by eilonhuppert — APPROVED** [review]

**Pull request merged**